app/models/host/managed.rb

```diff
@@ -557,10 +557,26 @@ def set_hostgroup_defaults(force = false)
   end
 
   def inherited_attributes
-    inherited_attrs = %w{domain_id operatingsystem_id architecture_id compute_resource_id}
+    inherited_attrs = %w{domain_id operatingsystem_id compute_resource_id}
     inherited_attrs << "subnet_id" unless compute_provides?(:ip)
     inherited_attrs << "subnet6_id" unless compute_provides?(:ip6)
-    inherited_attrs.concat(%w{medium_id ptable_id pxe_loader}) unless image_build?
+
+    if managed?
+      inherited_attrs << "architecture_id"
+
+      unless image_build?
+        inherited_attrs.concat(%w{ptable_id pxe_loader})
+
+        if hostgroup.medium_id.present?
```

Comment:
This would break the deployment / provisining using nested host groups.

- parent host group A has installation media.
- nested host group B is using host group A but doesn't define installation media.
- host is using host group B

I would suggest to drop line 570 - 576. HostRegistration would still work because of line 564.